docker/Dockerfile

```diff
@@ -31,7 +31,7 @@ ARG PYTHON_VERSION=3.9
 ###
 ### Stage 0: generate requirements.txt
 ###
-FROM docker.io/python:${PYTHON_VERSION}-slim as requirements
+FROM docker.io/python:${PYTHON_VERSION}-slim-bullseye as requirements
```

Comment:
Might be worth adding a comment here why we're using `bullseye` here now:

> Also enforce bullseye tag for the base Python images so we are sure to not have some discrepancy regarding glibc version among others.

Reply:
I don't really understand what you mean about glibc here @MatMaul. It's important that the builder and the runtime are consistent...but that's already the case isn't it, because they both used `docker.io/python:${PYTHON_VERSION}-slim` before?

Reply:
I'll add a rational along this one as a comment in the code as suggested.

Reply:
Blimey, we do copy a lot of binaries around. Comment sounds good, thanks